clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/op/job/KubectlJobExecutorTest.java

```diff
@@ -135,44 +181,251 @@ void topPodMultipleContainers() throws Exception {
     }
   }
 
+  @DisplayName("test to verify how kubectl errors are handled when retries are disabled")
   @Test
-  void kubectlRetryHandlingForConfiguredErrorsThatContinueFailingAfterMaxRetryAttempts() {
-    JobExecutor jobExecutor = mock(JobExecutor.class);
+  void kubectlJobExecutorErrorHandlingWhenRetriesAreDisabled() {
+    // when
     when(jobExecutor.runJob(any(JobRequest.class)))
         .thenReturn(
             JobResult.<String>builder()
                 .result(Result.FAILURE)
                 .output("")
-                .error("Unable to connect to the server: net/http: TLS handshake timeout")
+                .error("some error")
                 .build());
 
-    KubernetesConfigurationProperties kubernetesConfigurationProperties =
-        new KubernetesConfigurationProperties();
-    kubernetesConfigurationProperties.getJobExecutor().getRetries().setEnabled(true);
-    kubernetesConfigurationProperties.getJobExecutor().getRetries().setBackOffInMs(500);
-
     KubectlJobExecutor kubectlJobExecutor =
         new KubectlJobExecutor(
-            jobExecutor, "kubectl", "oauth2l", kubernetesConfigurationProperties);
+            jobExecutor, kubernetesConfigurationProperties, new SimpleMeterRegistry());
 
+    // then
     KubectlJobExecutor.KubectlException thrown =
         assertThrows(
             KubectlJobExecutor.KubectlException.class,
-            () -> kubectlJobExecutor.topPod(mockKubernetesCredentials(), "test", "test-pod"));
+            () -> kubectlJobExecutor.topPod(mockKubernetesCredentials(), "test", ""));
+
+    assertTrue(thrown.getMessage().contains("some error"));
+    // should only be called once as no retries are performed for this error
+    verify(jobExecutor).runJob(any(JobRequest.class));
+  }
+
+  @DisplayName(
+      "parameterized test to verify retry behavior for configured retryable errors that fail even after all "
+          + "attempts are exhausted")
+  @ParameterizedTest(
+      name = "{index} ==> number of simultaneous executions of the action under test = {0}")
+  @ValueSource(ints = {1, 10})
+  void kubectlRetryHandlingForConfiguredErrorsThatContinueFailingAfterMaxRetryAttempts(
+      int numberOfThreads) {
+    // setup
+    kubernetesConfigurationProperties.getJobExecutor().getRetries().setEnabled(true);
+
+    // to test log messages
+    Logger logger = (Logger) LoggerFactory.getLogger(KubectlJobExecutor.class);
+    ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+    listAppender.setContext((LoggerContext) LoggerFactory.getILoggerFactory());
+    logger.addAppender(listAppender);
+    listAppender.start();
```

Comment:
I assume you're already following an established testing pattern in this file, but in the future, note that Log4j2 (log4j-core-test or tests, I forget the full name) provides a `ListAppender` in its test code which has some additional methods useful for waiting on async things to finish logging, etc., along with the whole `@LoggerContextSource` JUnit 5 extension for using test-specific logging configurations.

Reply:
https://github.com/spinnaker/clouddriver/pull/5854/commits/4d9e4877a6f3190561c93c9dc57ee7c5a10ef345 uses [MemoryAppender](https://github.com/spinnaker/kork/blob/2244cc04c5d6cb09462a412168bc2a47d2eb5636/kork-test/src/main/java/com/netflix/spinnaker/kork/test/log/MemoryAppender.java) from kork, which seems like an improvement.  I need to learn about these other mechanisms since I bet they're even better.

In an even less exciting discovery, none of these junit tests in clouddriver-kubernetes are actually running.  https://github.com/spinnaker/clouddriver/pull/5855 takes care of that.